pyi.py

```diff
@@ -1087,11 +1087,41 @@ def _check_union_members(self, members: Sequence[ast.expr]) -> None:
                 self.error(member_list[1], Y016.format(unparse(member_list[1])))
 
         if not analysis.dupes_in_union:
+            self._check_for_Y051_violations(analysis)
             if analysis.multiple_literals_in_union:
                 self._error_for_multiple_literals_in_union(first_union_member, analysis)
             if not self.visiting_TypeAlias.active:
                 self._check_for_redundant_numeric_unions(first_union_member, analysis)
 
+    def _check_for_Y051_violations(self, analysis: UnionAnalysis) -> None:
+        """Search for redundant unions fitting the pattern `str | Literal["foo"]`, etc."""
+        literal_classes_present: defaultdict[str, list[_LiteralMember]]
+        literal_classes_present = defaultdict(list)
+        for literal in analysis.combined_literal_members:
+            if isinstance(literal, ast.Str):
+                literal_classes_present["str"].append(literal)
+            elif isinstance(literal, ast.Bytes):
+                literal_classes_present["bytes"].append(literal)
+            elif isinstance(literal, ast.Num):
+                if type(literal.n) is int:
+                    literal_classes_present["int"].append(literal)
+            elif isinstance(literal, ast.NameConstant):
+                if type(literal.value) is bool:
+                    literal_classes_present["bool"].append(literal)
+        for cls in ("str", "bytes", "int", "bool"):
```

Comment:
```suggestion
        for cls in literal_classes_present:
```
And then you can remove the condition on line 1114 below

Reply:
D'oh, of course